### PR TITLE
avoid unnecessary recompression in block protocol

### PR DESCRIPTION
**Pull request by arnetheduck**
Blocks can be sent straight from compressed data sources

### CODE REVIEW

File: beacon_chain/networking/eth2_network.nim [hunk 1]
```diff
@@ -533,6 +533,25 @@ proc isLightClientRequestProto(fn: NimNode): NimNode =
 
   return newLit(false)
 
+proc writeChunkSZ*(
+    conn: Connection, responseCode: Option[ResponseCode],
+    uncompressedLen: uint64, payloadSZ: openArray[byte],
+    contextBytes: openArray[byte] = []): Future[void] =
+  var output = memoryOutput(payloadSZ.len + contextBytes.len + 11)
```
Comment (etan-status) [inline beacon_chain/networking/eth2_network.nim]: Where is the 11 coming from?

File: beacon_chain/sync/sync_protocol.nim [hunk 2]
```diff
@@ -334,11 +336,16 @@ p2pProtocol BeaconSync(version = 1,
...
+        let uncompressedLen = uncompressedLenFramed(bytes).valueOr:
+          warn "Cannot read block size, database corrupt?",
+            bytes = bytes.len(), blck = shortLog(blockRef)
+          continue
+
         peer.updateRequestQuota(blockResponseCost)
         peer.awaitNonNegativeRequestQuota()
 
-        await response.writeRawBytes(bytes, []) # phase0 bytes
+        await response.writeBytesSZ(uncompressedLen, bytes, []) # phase0
```
Comment (etan-status) [inline beacon_chain/sync/sync_protocol.nim:348]:
```suggestion
        await response.writeBytesSZ(uncompressedLen, bytes, []) # phase0 bytes
```

(above at line 284, the comment was not deleted. I guess it got removed here unintentionally)

**Review by etan-status — APPROVED** [review]

File: beacon_chain/networking/eth2_network.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/networking/eth2_network.nim]: up to 10 bytes of leb128 and one byte of response code, worst case

Comment (etan-status) [reply beacon_chain/networking/eth2_network.nim]:
```suggestion
  const numOverheadBytes = sizeof(byte) + Leb128.maxLen(typeof(uncompressedLen))
  var output = memoryOutput(payloadSZ.len + contextBytes.len + numOverheadBytes)
```

**Pull request merged**